actors/test/actor-tests/src/test/java/cloud/orbit/actors/test/DeactivationTest.java

```diff
@@ -113,6 +113,25 @@ public void cleanupTest() throws ExecutionException, InterruptedException
         assertEquals(2, set.size());
     }
 
+    @Test
+    public void twoStagesDirtyCache() throws ExecutionException, InterruptedException, TimeoutException
+    {
+        clock.stop();
+        Stage stage1 = createStage();
+        Stage stage2 = createStage();
+        SomeActor actor1 = Actor.getReference(SomeActor.class, "1");
+
+        stage1.bind();
+        actor1.sayHello("huuhaa").join();
+        stage2.bind();
+        assertNotNull(actor1.sayFarewell().join());
+        clock.incrementTimeMillis(TimeUnit.HOURS.toMillis(1));
+        stage1.cleanup().join();
+        Thread.sleep(1500); // give stage2 time to process notification
```

Comment:
Remove and moved are one way messages so there is no reliable way to know when the message has been processed, we cannot poll the actor either as that would simply trigger activation.